qiskit/circuit/__init__.py

```diff
@@ -10,38 +10,62 @@
 # copyright notice, and modified files need to carry a notice indicating
 # that they have been altered from the originals.
 
-"""
-========================================
-Quantum Circuits (:mod:`qiskit.circuit`)
-========================================
+r"""
+=============================================
+Quantum circuit model (:mod:`qiskit.circuit`)
+=============================================
 
 .. currentmodule:: qiskit.circuit
 
-Overview
-========
+The fundamental element of quantum computing is the *quantum circuit*.  This is a computational
+routine that can be run, one shot at a time, on a quantum processing unit (QPU).  A circuit will act
+on a predefined amount of quantum data (in Qiskit, we only directly support qubits) with unitary
+operations (gates), measurements and resets.  In addition, a quantum circuit can contain operations
+on classical data, including near-time computations and control-flow constructs, which are executed
+by the controllers of the QPU.
```

Comment:
```suggestion
A fundamental model of quantum computations is the *quantum circuit*.  A quantum circuit represents a computational routine that can be run on a quantum processing unit (QPU).  A quantum circuit acts
on a predefined amount of quantum data with unitary operations (quantum gates), measurements and resets.  In addition, a quantum circuit can contain operations over classical and quantum data, including near-time computations and classical control flow constructs, which are executed by the classical control hardware 
of the QPU. A quantum circuit is typically executed multiple times on a QPU to gather statistics about the prepared quantum data. One such quantum circuit execution, including measurements, is called a*shot*. Qiskit natively supports quantum bits (qubits) as quantum data.
```

Reply:
"executed by the QPU control hardware" kind of _is_ the definition of "near-time compute"

Reply:
I think this point was largely resolved by the terminology switch to "real time", which is more immediately descriptive anyway.